packages/media-query/src/use-media-query.ts

```diff
@@ -54,14 +54,22 @@ export function useMediaQuery(query: string | string[]): boolean[] {
       // media query is matched. Using addEventListener on the window object
       // to listen for the resize event will call the callback on every
       // viewport resize.
-      mediaQueryList[index].addEventListener("change", listener)
+      if(typeof mediaQueryList[index].addEventListener === 'function') {
+        mediaQueryList[index].addEventListener("change", listener);
+      } else {
+        mediaQueryList[index].addListener(listener);
+      }
```

Comment:
This is a viable solution to define a polyfill for `addEventListener`, but I'd love to see this implementation somewhere else.
We maintain a `utils` package where the helper function `addDomEvent` is located.

https://github.com/chakra-ui/chakra-ui/blob/2133a4122672f23a5816d7010d8441067cba13a9/packages/utils/src/dom.ts#L63-L73

Kindly add the polyfill there and use the `addDomEvent` function here to attach the change listener.

Reply:
@TimKolberger 

the `addEventListener` on a DOM event is different from the MediaQueryList event

The util would look like:

```ts
export function addMediaQueryListEvent(
  type: "change",
  target: MediaQueryList,
  handler: (mqlEvent: MediaQueryListEvent) => void,
  options?: AddEventListenerOptions,
) {
  if(typeof target.addEventListener === 'function') {
    target.addEventListener(type, handler, options);
    return () => target.removeEventListener(type, handler, options);
  } 
  target.addListener(handler);
  return () => target.removeListener(handler)
}
```

but in the forEach, we can't use the return value of the `addMediaQueryListEvent`, since we are returning the listener.

So we have to define another util?

```ts
export function removeMediaQueryListEvent(
  type: "change",
  target: MediaQueryList,
  handler: (mqlEvent: MediaQueryListEvent) => void,
  options?: AddEventListenerOptions,
) {
  if(typeof target.removeEventListener === 'function') {
    target.removeEventListener(type, handler, options);
  } 
  target.removeListener(handler)
}
```

So it would be:

```ts
    addMediaQueryListEvent("change", mediaQueryList[index], listener)
    return listener
    })

   return () => {
     mediaQueryList.forEach((_, index) => {
      removeMediaQueryListEvent("change", mediaQueryList[index], listenerList[index])
    })
```

Reply:
Thank you for the detailled description!

TIL that this is only the case for `mediaQueryList` for Safari 14, not for all event listeners.

> Before Safari 14, MediaQueryList is based on EventTarget, so you must use addListener() and removeListener() to observe media query lists. - [MDN](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener)

This invalidates my suggestion and I think we can proceed with the current solution.

One last thing - kindly run prettier on this file to add a space between `if` and `(`.
I think then its good to go 🚀

Reply:
> Thank you for the detailled description!
> 
> TIL that this is only the case for `mediaQueryList` for Safari 14, not for all event listeners.
> 
> > Before Safari 14, MediaQueryList is based on EventTarget, so you must use addListener() and removeListener() to observe media query lists. - [MDN](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener)
> 
> This invalidates my suggestion and I think we can proceed with the current solution.
> 
> One last thing - kindly run prettier on this file to add a space between `if` and `(`. I think then its good to go 🚀

ahh right, that makes sense. 

I've run prettier 🚀